lib/src/client.dart

```diff
@@ -975,6 +962,19 @@ class Client extends MatrixApi {
           ));
         });
```

Comment:
I only noticed it now, but the room.state.forEach needs to be after the room.timeline.events.forEach for the setState, since the .state is the state events from before the timeline in the spec. Can you fix that too?